rsocket/statemachine/RequestResponseResponder.cpp

```diff
@@ -13,16 +13,24 @@ using namespace yarpl::flowable;
 
 void RequestResponseResponder::onSubscribe(
     Reference<yarpl::single::SingleSubscription> subscription) noexcept {
+#ifdef DEBUG
+  DCHECK(!gotOnSubscribe_.exchange(true)) << "Already called onSubscribe()";
+#endif
+
   if (StreamStateMachineBase::isTerminated()) {
     subscription->cancel();
     return;
   }
-  DCHECK(!producingSubscription_);
   producingSubscription_ = std::move(subscription);
 }
 
 void RequestResponseResponder::onSuccess(Payload response) noexcept {
-  DCHECK(producingSubscription_) << "didnt call onSubscribe";
+#ifdef DEBUG
```

Comment:
As we do DCHECK, why do we need DEBUG flag checking?

Reply:
So we're not compiling in the additional atomics in debug code, which would be a perf hit for opt builds as well

Reply:
DCHECK lines will get deleted, isn't it?

Reply:
They will, but DCHECKs won't be tested in opt code either, so we're not doing less bug checking compared to before

Reply:
What I mean is do we need the `#ifndef DEBUG #endif` ,, these two lines at all?
As the line, which has DCHECK will be immediately removed in a non-debug build, why do we need to protect that region with #ifndef DEBUG?

Reply:
The `DCHECK(x)` effectively compiles to `while(false) { CHECK(x) }`, so the expression `x` needs to be valid. `#define`ing out the atomics but not the DCHECKS would lead to "undefined symbol gotOnSubscribe_" type compilation errors.

Reply:
Cool. This is what I was asking for.